Notebook completed

alvgonfri opened the pull request:
We have finished the implementation of the Jupyter Notebook that contains the code of the clustering algorithms coparison.

alvarobernal2412 committed:
docs: Create readme.md

Initial version of readme have been included in develop branch.

alvgonfri committed:
git commit --amend
feat: clustering method & k-means implementation

The clustering method has been implemented and tested with k-means algorithm.

Refs #3
Refs #11

alvgonfri committed:
Clustering method & k-means implementation

The clustering method has been implemented and tested with k-means algorithm.

Refs #3
Refs #11

alvgonfri committed:
Merge branch 'develop' of https://github.com/alvgonfri/clustering_algorithms into develop

alvgonfri committed:
feat: visualizing a labelled dataset

A new function that visualizes a labelled dataset using a 3D plot has been implemented.

Refs #14

alvgonfri committed:
feat: quality metrics method

The quality metrics method has been implemented.

Refs #12

alvarobernal2412 committed:
feat: New methods added

New methods have been included. Refs: #10 #2 #8

alvgonfri committed:
feat: BIRCH implementation

BIRCH has been implemented and tested.

Refs #1

alvgonfri committed:
feat: Mean Shift implementation

Mean Shift has been implemented and tested.

Refs #9

alvgonfri committed:
feat: Spectral Clustering implementation

Mean Shift has been implemented and tested.

alvgonfri committed:
feat: PAM implementation

PAM has been implemented and tested.

alvarobernal2412 committed:
feat: UPGMA initial structure added

New algorithm have been included and a new one have been created. #Refs: #6 #4

alvarobernal2412 committed:
fix: UPGMA error

Small fixes done. Refs: #6

alvarobernal2412 committed:
feat: UPGMA implemented

The algorithm now correctly shows the figure. The index and the readme have been updated. Refs: #6

alvgonfri committed:
feat: new UPGMA implementation

A new UPGMA algorithm implementation option has been included.

Refs #6

alvgonfri committed:
feat: quality metrics computed

The quality metrics feature has been completed so that you can get the value of the metrics.

Refs #12

alvgonfri committed:
docs: notebook documentation improved

I have added a brief explanation of what we do in each section of the notebook.

alvarobernal2412 committed:
feat: New method added

A new method called algorithms_hyperparameters have been included to the notebook. Refs: #13

alvarobernal2412 committed:
feat: Optimal algorithm method correctly implemented

The method have been tested and needs to be refactored. Refs: #13

alvarobernal2412 committed:
feat: Optimal algorithm refactored

The algorithm method have been refactored. Refs: #15

alvgonfri committed:
feat: new UPGMA version

A new UPGMA algorithm version, which needs to be optimizated, has been included.

alvgonfri committed:
feat: new UPGMA version

A new function that tries to implement UPGMA algorithm has been created.

alvarobernal2412 committed:
feat: New plot method implemented

A new method has been created that shows the different values of the hiperparameters, with a algorithm given. Refs: #15

alvgonfri committed:
feat: New BIRCH parameters

I have added new parameters to the BIRCH algorithm in order to imporve its analysis.

alvgonfri committed:
fix: BIRCH small fix.

BIRCH small fix that improves the clustering results.

alvarobernal2412 committed:
feat: UPGMA new method started

A new implementation have been done, it is an initial version but it groups de clusters in tuples. Refs: #15

alvgonfri committed:
feat: UPGMA algorithm working

The UPGMA algorithm has been implemented.

alvarobernal2412 committed:
fix: Bar plot method finished

The bar plot method have been fixed to shows the algorithms with two hyperparameters. Refs: #13

alvgonfri committed:
feat: get optimal algorithm function

A function that gets the optimal algorithm for our dataset has been created.

alvgonfri committed:
docs: last section documentation

The methods used in the last section, "Optimal algorithm and hyperparameters", have beem described.

alvarobernal2412 committed:
docs: Update readme

Readme has been updated due to the proyect development. Refs: None

alvarobernal2412 committed:
docs: Update readme

New image added to the readme. Refs: None

alvarobernal2412 committed:
docs: Update readme.md

Small fix in the readme. Refs: None

alvarobernal2412 reviewed (changes requested):
Clustering5.pdf must be added before merging.

alvarobernal2412 committed:
docs: Update readme.md

Small correction made in the readme. Refs:None

alvgonfri committed:
docs: documentation added

The file containing the documentation, called `ClusteringG5.pdf`, has been included.

alvarobernal2412 commented:
After the final review, the requested changes and some corrections have been successfully made. Finally, the changes can be incorporated into the main version.
